### PR TITLE
I-641 Added feedback on reprojection + creating work unit from fits file

### PR DESCRIPTION
**Pull request by vlnistor**
This PR aims to tackle #641. I use `tqdm` package for feedback

1. Added feedback during reprojection
2. Added feedback during the creation of a `WorkUnit` from `fits` file

### CODE REVIEW

File: src/kbmod/work_unit.py [hunk 1]
```diff
@@ -280,7 +285,11 @@ def from_fits(cls, filename):
             per_image_wcs = []
             per_image_ebd_wcs = []
             constituent_images = []
-            for i in range(n_constituents):
+            for i in tqdm(
```
Comment (vlnistor) [inline src/kbmod/work_unit.py:288]: I'm assuming the process that takes long is the one that creates the `LayeredImage` (above). Is there a need for feedback on this as well?

**Comment by vlnistor** [comment]
When it comes to trying to add feedback on creating a `WorkUnit` from an `ImageCollection` I couldn't get an example to work:
1. `discrete_piles_e2e.ipynb` contains an example but requires the `lsst` stack
2. `test_imagecollection.py` has a set of mock `ImageCollection`s but they throw when calling `toWorkUnit()` on them

**Comment by vlnistor** [comment]
@maxwest-uw ,

- [x] https://github.com/dirac-institute/kbmod/blob/main/src/kbmod/work_unit.py#L288
- [x] https://github.com/dirac-institute/kbmod/blob/main/src/kbmod/reprojection.py#L129
- [x] https://github.com/dirac-institute/kbmod/blob/main/src/kbmod/reprojection.py#L271 
- [ ]  https://github.com/dirac-institute/kbmod/blob/main/src/kbmod/image_collection.py#L531 - Unclear what you meant here

Testing 

- [x] Test piping the output to a file and make sure that we can read the TQDM output nicely when not running directly from the command line - You need to redirect `stderror` for `tqdm` output to appear. It looks similar to this

```
Setting unknown parameter: cluster_function
Setting unknown parameter: num_cores

Loading images:   0%|          | 0/4 [00:00]
Loading images: 100%|██████████| 4/4 [00:00]

Reprojecting:   0%|          | 0/3 [00:00]
Reprojecting: 100%|██████████| 3/3 [00:00]

```

**Comment by vlnistor** [comment]
### Todo

- [x] https://github.com/dirac-institute/kbmod/blob/main/src/kbmod/work_unit.py#L288
- [x] https://github.com/dirac-institute/kbmod/blob/main/src/kbmod/reprojection.py#L129
- [x] https://github.com/dirac-institute/kbmod/blob/main/src/kbmod/reprojection.py#L271 - 

    I played with the `reproject_demo` notebook. I added a `time.sleep(3)` inside [`_reproject_images`](https://github.com/dirac-institute/kbmod/blob/84f46da2d09dec93c19ebb14b3ec53435f65f138/src/kbmod/reprojection.py#L394). For a work unit with 3 [`unique_obstimes_indices`](https://github.com/dirac-institute/kbmod/blob/84f46da2d09dec93c19ebb14b3ec53435f65f138/src/kbmod/reprojection.py#L257), and [`MAX_PROCESSES`](https://github.com/dirac-institute/kbmod/blob/84f46da2d09dec93c19ebb14b3ec53435f65f138/src/kbmod/reprojection.py#L11C1-L11C14) set to 1, the progress bar did 3 hops every 3 seconds (as expected). When you set the `MAX_PROCESSES` to 8 (above the number of parallell jobs) the progress bars hops once (after 3 seconds) as there were enough threads to do the jobs in parallel (again, as expected).
- [ ]  https://github.com/dirac-institute/kbmod/blob/main/src/kbmod/image_collection.py#L531 - Unclear what you meant here

### Testing 

- [x] Test piping the output to a file and make sure that we can read the TQDM output nicely when not running directly from the command line - You need to redirect `stderror` for `tqdm` output to appear (will that be an issue?). It looks similar to this

```
Setting unknown parameter: cluster_function
Setting unknown parameter: num_cores

Loading images:   0%|          | 0/4 [00:00]
Loading images: 100%|██████████| 4/4 [00:00]

Reprojecting:   0%|          | 0/3 [00:00]
Reprojecting: 100%|██████████| 3/3 [00:00]

```

**Note** This PR also contains progress update on loading from the `fits` file. This wasn't in the original todo. Should I remove?

**Comment by maxwest-uw** [comment]
I think the line numbers may have gotten shifted around, sorry, there should also be `tqdm` on [this](https://github.com/dirac-institute/kbmod/blob/main/src/kbmod/work_unit.py#L283) line as well in work_unit.py, the loop
```
for i in range(num_images):
                sci_hdu = hdul[f"SCI_{i}"]

                # Read in the layered image from different extensions.
                img = LayeredImage(
                    sci_hdu.data.astype(np.single),
                    hdul[f"VAR_{i}"].data.astype(np.single),
                    hdul[f"MSK_{i}"].data.astype(np.single),
                    PSF(hdul[f"PSF_{i}"].data),
                    sci_hdu.header["MJD"],
                )
```

which is the one that actually loads in the images. Everything else looks good!

**Comment by vlnistor** [comment]
@maxwest-uw that should all be pushed now. Let me know if anything else

**Review by maxwest-uw — APPROVED** [review]
lgtm !

**Pull request merged**